File: 06Adapter/02TwoWayAdapter/Seacraft.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _02TwoWayAdapter {
    // Adaptee implementation
    public class Seacraft : ISeacraft {
        int speed = 0;

        public virtual void IncreaseRevs() {
            speed += 10;
            Console.WriteLine("Seacraft engine increases revs to " + speed + " knots");
        }
        public int Speed
        {
            get { return speed; }
        }
    }

}

File: 06Adapter/Students/2019/PawelskiBartlomiej/twoWayAdapter/Seacraft.java
package twoWayAdapter;

public class Seacraft implements ISeacraft {
    private int speed = 0;

    @Override
    public int speed() {
        return speed;
    }

    @Override
    public void increaseRevs() {
        speed += 10;
        System.out.println("Seacraft engine increases revs to " + speed + " knots");
    }
}
